bimsb/packages/bioinformatics-nonfree.scm

```diff
@@ -1882,3 +1883,47 @@ MACE is a bioinformatics tool dedicated to analyze ChIP-exo data.")
     ;; others say "free for all use", others say "GPL version 1.3" (?), yet
     ;; others say "Artistic license".
     (license license:gpl2+)))
+
+;; Although this program is released under the GPL it depends on
+;; ViennaRNA and bowtie1, which are non-free software.
+(define-public mirdeep2
+  (package
+    (name "mirdeep2")
+    (version "0.0.92")
+    (source (origin
+              (method git-fetch)
+              (uri (git-reference
+                   (url "https://github.com/rajewsky-lab/mirdeep2.git")
+                   (commit (string-append "v" version))))
+              (file-name (string-append name "-" version "-checkout"))
+              (sha256
+               (base32
+                "1gasx6vb38n4vbjbn5s7ch2rpflgf2crdd8mdp2xm571w2j01wmm"))))
+    (build-system gnu-build-system)
+    (arguments
+     `(#:phases
+       (modify-phases %standard-phases
+         (delete 'configure)
+         (delete 'build)
+         (replace 'install
+           (lambda* (#:key inputs outputs #:allow-other-keys)
+             (let ((target (string-append (assoc-ref outputs "out")
+                                          "/bin")))
+               (with-directory-excursion "src"
+                 (copy-recursively "." target))
+               (with-output-to-file (string-append target "/../install_successful")
```

Comment:
@rekado: Based on your comment in the first round of review

> you can avoid `(chdir "foo") ... (bar "../baz")` by using `(with-directory-excursion "foo" ...) ... (bar "baz")`.

I think you understood that the `../` does referred to the `src` directory.
Note that this is not the case. That's also the reason this line is not part of the `(with-directory-excursion ...)` expression.

miRDeep2 comes with a `perl` script to download all dependencies, patching their Makefiles and installing them locally, and modifying `.bashrc` or `.zshrc` to set the `PATH` and `PERL5LIB`. In the end, it touches a file that is checked for *by the binaries at runtime* to ensure mirdeep2 was 'properly' installed. As I wanted the dependencies and path modifications (or shebang-patching) to be handled by Guix, I did not use this script but followed the manual build instructions (adjusting on the way as needed). However, I need to make sure this file exists which is expected (hardcoded) one level *above* the `bin` directory containing the binaries (note that AFAIK, this directory doesn't need to be called `bin` but *must have a three character name*). That's why I append `/../` to `target` but if you prefer another way (i.e. some Scheme version of `dirname` or `(string-append (assoc-ref outputs "out") "/install_successful")`, let me know.
Also note that this will symlink an empty file as `~/.guix-profile/install_successful` which is weird but I guess won't ever introduce any actual conflict with another package. :wink:

I just wanted to clarify this to get your opinion because this is probably the weirdest version of `make check` I've ever come across. :laughing: